numba/ir_utils.py

```diff
@@ -1926,6 +1926,12 @@ def raise_on_unsupported_feature(func_ir, typemap):
                     isinstance(ty, types.DictType)):
                     raise TypingError(msg % (ty, stmt.value.name, ty), loc=stmt.loc)
 
+            # checks for generator expressions (yield in use when func_ir has
+            # not been identified as a generator).
+            if isinstance(stmt.value, ir.Yield) and not func_ir.is_generator:
```

Comment:
There should be a similar check in `InlineClosureCallPass` so that it either:

* reject inlining if it will produce an illegal IR; or,
* abort compilation after it produces an illegal IR.

Reply:
Good point, will do the former as the inlining target has to be walked anyway.